File: Scripts/Core/Simulation/PlayByPlay/EventBuilder.cs
using System;
using System.Collections.Generic;
using HardwoodHoops.Core.Domain.Games;
using DomainGameEvent = HardwoodHoops.Core.Domain.Games.GameEvent;

namespace HardwoodHoops.Core.Simulation.PlayByPlay;

public sealed class EventBuilder
{
    public DomainGameEvent Build(string eventType, string description, Dictionary<string, object> payload)
    {
        var timestamp = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
        return new DomainGameEvent(eventType, description, payload, timestamp);
    }
}

File: Next-Up-Hardwood-Hoops-main/courthoops/simulation/playbyplay/event_builder.py
import time
from typing import Any, Dict

from courthoops.domain.game.events import Event


class EventBuilder:
    """Creates domain event objects from raw simulation data."""

    def build(self, event_type: str, description: str, payload: Dict[str, Any]) -> Event:
        return Event(event_type=event_type, description=description, payload=payload, timestamp=time.time())


